### PR TITLE
Update to 6.1.1 and latest snapshot

### PR DESCRIPTION
**Pull request by timyates**

### CODE REVIEW

File: tracing-opentelemetry/build.gradle [hunk 1]
```diff
@@ -28,16 +31,18 @@ dependencies {
...
     testAnnotationProcessor projects.tracingOpentelemetryAnnotation
 
-//    testImplementation(libs.opentelemetry.contrib.aws.resources)
-    testImplementation(libs.opentelemetry.sdk.extension.aws)
+    testImplementation(libs.managed.opentelemetry.contrib.aws.resources)
+//    testImplementation(libs.opentelemetry.sdk.extension.aws)
 
-    testImplementation mn.reactor
-    testImplementation mn.micronaut.reactor.http.client
+    testImplementation mnReactor.micronaut.reactor
```
Comment (timyates) [inline tracing-opentelemetry/build.gradle:37]: Nitpick: I think this can stay as `mn.reactor`

File: tracing-opentelemetry-http/build.gradle [hunk 2]
```diff
@@ -19,10 +20,11 @@ dependencies {
     compileOnly libs.kotlinx.coroutines.core
     compileOnly libs.kotlinx.coroutines.reactor
 
-    testImplementation mn.reactor
-    testImplementation mn.micronaut.reactor.http.client
+    testImplementation mnReactor.micronaut.reactor
```
Comment (timyates) [inline tracing-opentelemetry-http/build.gradle:23]: I think this can remain as `mn.reactor`

File: tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCSpec.groovy [hunk 3]
```diff
@@ -34,7 +34,8 @@ class MDCSpec extends Specification {
     @Shared
     @AutoCleanup
     EmbeddedServer embeddedServer = ApplicationContext.run(EmbeddedServer, [
-            'mdc.test.enabled': true
+            'mdc.test.enabled': true,
+            'micronaut.http.client.read-timeout' : '30s'
```
Comment (timyates) [inline tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCSpec.groovy:38]: Was this to aid debugging?  (Same for MDCSpec2 and ReactorRx2JavaSpec)

**Review by timyates — COMMENTED** [review]
Nice!  Couple of comments, and 5 of the [Code Smells](https://sonarcloud.io/project/issues?id=micronaut-projects_micronaut-tracing&pullRequest=213&resolved=false&types=CODE_SMELL) could be easily addressed?

File: tracing-opentelemetry/build.gradle (hunk 1 above)
Comment (n0tl3ss) [reply tracing-opentelemetry/build.gradle:37]: it has to be `mnReactor.micronaut.reactor` because some code is moved from micronaut-core to micronaut-reactor

File: tracing-opentelemetry-http/build.gradle (hunk 2 above)
Comment (n0tl3ss) [reply tracing-opentelemetry-http/build.gradle:23]: it has to be `mnReactor.micronaut.reactor` because some code is moved from micronaut-core to micronaut-reactor

File: gradle.properties [hunk 4]
```diff
@@ -2,8 +2,8 @@ projectVersion=5.0.0-SNAPSHOT
 projectGroup=io.micronaut.tracing
 micronautDocsVersion=2.0.0
 micronautVersion=4.0.0-SNAPSHOT
-micronautTestVersion=3.7.0
-groovyVersion=4.0.5
+micronautTestVersion=4.0.0-SNAPSHOT
```
Comment (sdelamo) [inline gradle.properties:5]: can we get rid of this and only have `libs.versions.toml` ?

File: gradle.properties [hunk 5]
```diff
@@ -2,8 +2,8 @@ projectVersion=5.0.0-SNAPSHOT
 projectGroup=io.micronaut.tracing
 micronautDocsVersion=2.0.0
 micronautVersion=4.0.0-SNAPSHOT
```
Comment (sdelamo) [inline gradle.properties:4]: can we get rid of this and only have `libs.versions.toml` ?

File: gradle.properties [hunk 6]
```diff
@@ -2,8 +2,8 @@ projectVersion=5.0.0-SNAPSHOT
 projectGroup=io.micronaut.tracing
 micronautDocsVersion=2.0.0
```
Comment (sdelamo) [inline gradle.properties:3]: can we get rid of this and only have `libs.versions.toml` ?

File: gradle/license.gradle [hunk 7]
```diff
@@ -0,0 +1,22 @@
+apply plugin: 'com.github.hierynomus.license'
```
Comment (sdelamo) [inline gradle/license.gradle:1]: why are we adding this file? 

project template does not have it https://github.com/micronaut-projects/micronaut-project-template/tree/master/gradle

File: src/main/docs/guide/opentelemetry/annotations.adoc [hunk 8]
```diff
@@ -3,4 +3,4 @@ The pkg:tracing.opentelemetry.processing[] package contains transformers and map
 To enable Open telemetry annotations you have to add next annotation processor in your dependency block:
 dependency:micronaut-tracing-opentelemetry-annotation[scope="annotationProcessor", version="{version}", groupId="io.micronaut.tracing"]
 
-The Open Tracing annotations that are defined inside the The pkg:tracing.annotation[] package are also available for usage inside Open Telemetry.
```
Comment (sdelamo) [inline src/main/docs/guide/opentelemetry/annotations.adoc:6]: can we backport this fix to 4.5.x ?

File: src/main/docs/guide/opentelemetry/kafka.adoc [hunk 9]
```diff
@@ -0,0 +1,23 @@
+To enable creating span objects on the every HTTP server request, client request, server response and client response you have to add next depedency:
```
Comment (sdelamo) [inline src/main/docs/guide/opentelemetry/kafka.adoc:1]: can we back port this to 4.5.x or only applies to MN 4?

File: src/main/docs/guide/toc.yml [hunk 10]
```diff
@@ -11,6 +11,7 @@ opentelemetry:
   idgenerator: ID Generator
   http: HTTP Server and Client
   grpc: gRPC Server and Client
+  kafka: Kafka
```
Comment (sdelamo) [inline src/main/docs/guide/toc.yml:14]: can we backport this to 4.5.x or only applies to MN 4?

File: src/main/docs/guide/zipkin.adoc [hunk 11]
```diff
@@ -61,8 +61,8 @@ tracing:
     enabled: true
     http:
       urls:
-        - http://foo:9411
-        - http://bar:9411
+        - https://foo:9411
+        - https://bar:9411
```
Comment (sdelamo) [inline src/main/docs/guide/zipkin.adoc:65]: can we backport this to 4.5.x or only applies to MN 4?

File: tests/kotlin-tests/src/test/kotlin/reactor/ReactorContextPropagationSpec.kt [hunk 12]
```diff
@@ -81,6 +82,7 @@ class TestController(private val someService: SomeService) {
 }
 
 @Introspected
+@Serdeable
```
Comment (sdelamo) [inline tests/kotlin-tests/src/test/kotlin/reactor/ReactorContextPropagationSpec.kt:85]: `@Serdeable` applies also `@Introspected`. Please remove `@Introspected` annotation.

File: tests/kotlin-tests/src/test/kotlin/reactor/ReactorContextPropagationSpec.kt [hunk 13]
```diff
@@ -147,4 +149,5 @@ suspend fun ServerFilterChain.next(request: HttpRequest<*>): MutableHttpResponse
...
```
Comment (sdelamo) [inline tests/kotlin-tests/src/test/kotlin/reactor/ReactorContextPropagationSpec.kt:152]: `@Serdeable` applies also `@Introspected`. Please remove `@Introspected` annotation.

File: tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCReactorSpec.groovy [hunk 14]
```diff
@@ -46,7 +46,7 @@ class MDCReactorSpec extends Specification {
     @Shared
     @AutoCleanup
     EmbeddedServer embeddedServer = ApplicationContext.run(EmbeddedServer, [
-            'mdc.reactortest.enabled': true
+            'mdc.reactortest.enabled': 'true',
```
Comment (sdelamo) [inline tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCReactorSpec.groovy:49]: use `StringUtils.TRUE` instead

File: tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCReactorSpec.groovy [hunk 15]
```diff
@@ -55,7 +55,7 @@ class MDCReactorSpec extends Specification {
 
     void "test MDC propagates"() {
         expect:
-        List<Tuple2> result = Flux.range(1, 100)
+        List<Tuple2> result = Flux.range(1, 2)
```
Comment (sdelamo) [inline tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCReactorSpec.groovy:58]: why the change from 100 to 2 ?

File: tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCReactorSpec.groovy [hunk 16]
```diff
@@ -76,7 +76,7 @@ class MDCReactorSpec extends Specification {
     }
 
     @Controller("/mdc")
-    @Requires(property = 'mdc.reactortest.enabled')
+    @Requires(property = 'mdc.reactortest.enabled', value = "true")
```
Comment (sdelamo) [inline tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCReactorSpec.groovy:79]: replace `value = "true"` with `value = StringUtils.TRUE`. Why do we need to set it in MN 4?

File: tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCReactorSpec.groovy [hunk 17]
```diff
@@ -140,7 +140,7 @@ class MDCReactorSpec extends Specification {
     }
 
     @Client("/mdc")
-    @Requires(property = 'mdc.reactortest.enabled')
+    @Requires(property = 'mdc.reactortest.enabled', value = "true")
```
Comment (sdelamo) [inline tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCReactorSpec.groovy:143]: replace `value = "true"` with `value = StringUtils.TRUE`. Why do we need to set it in MN 4?

File: tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCReactorSpec.groovy [hunk 18]
```diff
@@ -154,7 +154,7 @@ class MDCReactorSpec extends Specification {
     }
 
     @Filter(MATCH_ALL_PATTERN)
-    @Requires(property = 'mdc.reactortest.enabled')
+    @Requires(property = 'mdc.reactortest.enabled', value = "true")
```
Comment (sdelamo) [inline tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCReactorSpec.groovy:157]: replace `value = "true"` with `value = StringUtils.TRUE`. Why do we need to set it in MN 4?

File: tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCReactorSpec.groovy [hunk 19]
```diff
@@ -167,7 +167,7 @@ class MDCReactorSpec extends Specification {
     }
 
     @Filter("/mdc/test**")
-    @Requires(property = 'mdc.reactortest.enabled')
+    @Requires(property = 'mdc.reactortest.enabled', value = "true")
```
Comment (sdelamo) [inline tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCReactorSpec.groovy:170]: replace `value = "true"` with `value = StringUtils.TRUE`. Why do we need to set it in MN 4?

File: tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCSpec.groovy [hunk 20]
```diff
@@ -34,7 +34,7 @@ class MDCSpec extends Specification {
     @Shared
     @AutoCleanup
     EmbeddedServer embeddedServer = ApplicationContext.run(EmbeddedServer, [
-            'mdc.test.enabled': true
+            'mdc.test.enabled': true,
```
Comment (sdelamo) [inline tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCSpec.groovy:37]: the trailing comma is unnecessary

File: tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCSpec2.groovy [hunk 21]
```diff
@@ -28,7 +28,7 @@ class MDCSpec2 extends Specification {
     @Shared
     @AutoCleanup
     EmbeddedServer embeddedServer = ApplicationContext.run(EmbeddedServer, [
-            'mdc.test2.enabled': true
+            'mdc.test2.enabled': true,
```
Comment (sdelamo) [inline tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCSpec2.groovy:31]: the trailing comma is unnecessary

File: tracing-jaeger/src/test/groovy/io/micronaut/tracing/jaeger/JaegerTracerFactorySpec.groovy [hunk 22]
```diff
@@ -2,6 +2,7 @@ package io.micronaut.tracing.jaeger
 
 import io.jaegertracing.internal.JaegerTracer
 import io.micronaut.context.ApplicationContext
+import io.micronaut.runtime.server.EmbeddedServer
```
Comment (sdelamo) [inline tracing-jaeger/src/test/groovy/io/micronaut/tracing/jaeger/JaegerTracerFactorySpec.groovy:5]: remove unused import

File: tracing-opentelemetry-grpc/src/main/java/io/micronaut/tracing/opentelemetry/instrument/grpc/GrpcClientTracingInterceptorFactory.java [hunk 23]
```diff
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.micronaut.tracing.grpc;
+package io.micronaut.tracing.opentelemetry.instrument.grpc;
```
Comment (sdelamo) [inline tracing-opentelemetry-grpc/src/main/java/io/micronaut/tracing/opentelemetry/instrument/grpc/GrpcClientTracingInterceptorFactory.java:16]: this is a breaking change, can we document in `breaks.adoc` the package change?

File: tracing-opentelemetry-grpc/src/main/java/io/micronaut/tracing/opentelemetry/instrument/grpc/GrpcServerTracingInterceptorFactory.java [hunk 24]
```diff
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.micronaut.tracing.grpc;
+package io.micronaut.tracing.opentelemetry.instrument.grpc;
```
Comment (sdelamo) [inline tracing-opentelemetry-grpc/src/main/java/io/micronaut/tracing/opentelemetry/instrument/grpc/GrpcServerTracingInterceptorFactory.java:16]: this is a breaking change, can we document in `breaks.adoc` the package change?

File: tracing-opentelemetry-grpc/src/main/java/io/micronaut/tracing/opentelemetry/instrument/grpc/package-info.java [hunk 25]
```diff
@@ -21,7 +21,7 @@
 */
 @Requires(classes = {OpenTelemetry.class, GrpcTelemetry.class})
 @Requires(beans = OpenTelemetry.class)
-package io.micronaut.tracing.grpc;
+package io.micronaut.tracing.opentelemetry.instrument.grpc;
```
Comment (sdelamo) [inline tracing-opentelemetry-grpc/src/main/java/io/micronaut/tracing/opentelemetry/instrument/grpc/package-info.java:24]: this is a breaking change, can we document in `breaks.adoc` the package change?

File: tracing-opentelemetry-http/src/main/java/io/micronaut/tracing/opentelemetry/instrument/http/server/package-info.java [hunk 26]
```diff
@@ -15,7 +15,12 @@
  */
 /**
  * Classes related to the creation of span objects on the every HTTP server request and server response.
+ *
  * @author Sergio del Amo
  * @since 4.2.0
  */
+@Requires(classes = MutableHttpRequest.class)
```
Comment (sdelamo) [inline tracing-opentelemetry-http/src/main/java/io/micronaut/tracing/opentelemetry/instrument/http/server/package-info.java:22]: should we backport this to 4.5.x?

File: tracing-opentelemetry-kafka/build.gradle [hunk 27]
```diff
@@ -0,0 +1,29 @@
...
+
+    testImplementation mnSerde.micronaut.serde.jackson
+    testImplementation libs.opentelemetry.sdk
+    testImplementation libs.opentelemetry.sdk.testing
+    testImplementation libs.testcontainers.kafka
+    testImplementation libs.testcontainers.spock
+
+}
+
+// TODO temporarily disable binary compatibility checks
```
Comment (sdelamo) [inline tracing-opentelemetry-kafka/build.gradle:24]: please remove

File: tracing-opentelemetry/build.gradle [hunk 28]
```diff
@@ -28,16 +31,18 @@ dependencies {
 
     testAnnotationProcessor projects.tracingOpentelemetryAnnotation
 
-//    testImplementation(libs.opentelemetry.contrib.aws.resources)
-    testImplementation(libs.opentelemetry.sdk.extension.aws)
+    testImplementation(libs.managed.opentelemetry.contrib.aws.resources)
+//    testImplementation(libs.opentelemetry.sdk.extension.aws)
```
Comment (sdelamo) [inline tracing-opentelemetry/build.gradle:35]: why is this commented out?

**Review by sdelamo — CHANGES_REQUESTED** [review]

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCReactorSpec.groovy [hunk 29]
```diff
@@ -53,6 +54,7 @@ class MDCReactorSpec extends Specification {
     @AutoCleanup
     HttpClient client = HttpClient.create(embeddedServer.URL)
 
+    @Ignore("https://github.com/micronaut-projects/micronaut-core/issues/8617")
```
Comment (sdelamo) [inline tracing-core/src/test/groovy/io/micronaut/tracing/instrument/util/MDCReactorSpec.groovy:57]: @n0tl3ss  can you change this to` @PendingFeature` instead? That way, when #8167 is fixed the test will start passing and it will break the build and we will remember to remove `@PendingFeature`

**Review by sdelamo — CHANGES_REQUESTED** [review]
